src/scene/layer-composition.js

```diff
@@ -140,16 +140,14 @@ LayerComposition.prototype._update = function () {
         result |= COMPUPDATED_INSTANCES;
         this._meshInstances.length = 0;
         this._meshInstancesSet.clear();
-        var dirtyBlend1, dirtyBlend2;
 
         for (i = 0; i < len; i++) {
             layer = this.layerList[i];
             if (!layer.passThrough) {
 
                 // add meshInstances from both opaque and transparent lists
-                dirtyBlend1 = addUniqueMeshInstance(this._meshInstances, this._meshInstancesSet, layer.opaqueMeshInstances);
-                dirtyBlend2 = addUniqueMeshInstance(this._meshInstances, this._meshInstancesSet, layer.transparentMeshInstances);
-                this._dirtyBlend = dirtyBlend1 || dirtyBlend2;
+                this._dirtyBlend = addUniqueMeshInstance(this._meshInstances, this._meshInstancesSet, layer.opaqueMeshInstances) || this._dirtyBlend;
```

Comment:
Uhm. Shouldn't this be:
```
this._dirtyBlend = addUniqueMeshInstance(this._meshInstances, this._meshInstancesSet, layer.opaqueMeshInstances);
```

Reply:
No it can't. `addUniqueMeshInstance()` must be executed but `||=` operator short circuits.

Reply:
Ah ok